### PR TITLE
fix: hash navigation if items added in different render

### PR DESCRIPTION
**Pull request by nohardcode**

### CODE REVIEW

**Comment by nohardcode** [comment]
related #276

File: src/gallery-custom.tsx [hunk 1]
```diff
@@ -144,9 +148,26 @@ export const CustomGallery: FC<CustomGalleryProps> = ({
...
-  }, [])
+  const set = useCallback(
+    (ref, data: InternalItem) => {
+      const { id } = data
+      items.current.set(ref, data)
+
+      if (!pidRef.current) return
+
+      if (id === pidRef.current) {
+        open(ref)
```
Comment (irudoy) [inline src/gallery-custom.tsx:159]: `pidRef.current = null`

File: src/gallery-custom.tsx [hunk 2]
```diff
@@ -135,6 +134,11 @@ export const CustomGallery: FC<CustomGalleryProps> = ({
 
     const { pid, gid } = params
 
+    if (items.current.size === 0) {
```
Comment (irudoy) [inline src/gallery-custom.tsx:137]: need some test cases for it

File: src/gallery-custom.tsx [hunk 3]
```diff
@@ -42,6 +42,7 @@ export const CustomGallery: FC<CustomGalleryProps> = ({
   onOpen,
 }) => {
   const items = useRef(new Map<ItemRef, InternalItem>())
+  const pidRef = useRef(null)
```
Comment (irudoy) [inline src/gallery-custom.tsx:45]: need a more expressive name for this

maybe `enqueuedForOpeningPidRef`?
or `openWhenReadyPidRef`?

or something more compact and still meaningful enough, at your discretion

**Review by irudoy — APPROVED** [review]

**Pull request merged**